src/js/components/Button/Button.js

```diff
@@ -402,6 +436,46 @@ const Button = forwardRef(
       contents = <Badge content={badgeProp}>{contents}</Badge>;
     }
 
+    if (busy || success) {
+      // match what the label will use
+      let animationColor;
+      if (kind) {
+        if (!plain) {
+          animationColor =
+            (hover && getIconColor(themePaths.hover, theme)) ||
+            getIconColor(themePaths.base, theme, color, kind);
+        }
+      } else if (primary) {
+        animationColor =
+          theme.global.colors.text[isDarkBackground() ? 'dark' : 'light'];
+      }
+
+      const RelativeBox = styled(Box)`
```

Comment:
This should be outside of the component, perhaps around line 39?

Reply:
Fixed